test/ipc_negative.cpp

```diff
@@ -0,0 +1,49 @@
+#include "base.hpp"
+#include "pool_null.h"
+#include "provider_null.h"
+
+#include <umf/ipc.h>
+#include <umf/memory_pool.h>
+#include <umf/memory_provider.h>
+
+struct IpcNotSupported : umf_test::test {
+  protected:
+    void SetUp() override {
+        umf_memory_provider_ops_t provider_ops = UMF_NULL_PROVIDER_OPS;
+        provider_ops.ipc.get_ipc_handle_size = nullptr;
+        provider_ops.ipc.get_ipc_handle = nullptr;
+        provider_ops.ipc.open_ipc_handle = nullptr;
+        provider_ops.ipc.put_ipc_handle = nullptr;
+        provider_ops.ipc.close_ipc_handle = nullptr;
+
+        umf_result_t ret;
+        ret = umfMemoryProviderCreate(&provider_ops, nullptr, &provider);
+        ASSERT_EQ(ret, UMF_RESULT_SUCCESS);
+
+        ret = umfPoolCreate(&UMF_NULL_POOL_OPS, provider, nullptr,
+                            UMF_POOL_CREATE_FLAG_OWN_PROVIDER, &pool);
+        ASSERT_EQ(ret, UMF_RESULT_SUCCESS);
+    }
+
+    umf_memory_provider_handle_t provider;
+    umf_memory_pool_handle_t pool;
+};
+
+TEST_F(IpcNotSupported, GetIPCHandleSizeNotSupported) {
+    size_t size;
+    auto ret = umfPoolGetIPCHandleSize(pool, &size);
+    EXPECT_EQ(ret, UMF_RESULT_ERROR_NOT_SUPPORTED);
+}
+
+TEST_F(IpcNotSupported, OpenIPCHandleNotSupported) {
+    umf_ipc_handle_t ipc_handle = {};
+    void *ptr;
+    auto ret = umfOpenIPCHandle(pool, ipc_handle, &ptr);
+    EXPECT_EQ(ret, UMF_RESULT_ERROR_NOT_SUPPORTED);
+}
+
+TEST_F(IpcNotSupported, CloseIPCHandleInvalid) {
+    int local_var;
+    auto ret = umfCloseIPCHandle(&local_var);
+    EXPECT_EQ(ret, UMF_RESULT_ERROR_INVALID_ARGUMENT);
```

Comment:
In fact, if you pass some pointer that was not result of the `umfOpenIPCHandle` function then you get `UMF_RESULT_ERROR_INVALID_ARGUMENT` error. So with this test you are not checking if UMF tries to call `provider_ops.ipc.close_ipc_handle` callback, the UMf fails earlier because the argument is invalid.

Reply:
Yes, that was the idea - that case was missing for coverage. I see how it wasn't clear there so I have renamed test and added a comment.

Reply:
in `ipcFixture.cpp` we already have `TEST_P(umfIpcTest, GetIPCHandleSizeInvalidArgs)` and `TEST_P(umfIpcTest, GetIPCHandleInvalidArgs)` to test invalid args with other IPC APIs. Maybe we should move this particular test there?